File: ios/Classes/Adaptors/IASStatisticsManagerAdaptor.swift
import Flutter
import Foundation
@_spi(QAApp) import InAppStorySDK
@_spi(IAS_API) import InAppStorySDK

class IASStatisticsManagerAdaptor: InAppStoryStatManagerHostApi {
    func sendStatistics(
        enabled: Bool,
        completion: @escaping (Result<Void, any Error>) -> Void
    ) {
        InAppStory.shared.isStatisticDisabled = !enabled

        completion(.success(()))
    }

    init(binaryMessenger: FlutterBinaryMessenger) {
        InAppStoryStatManagerHostApiSetup.setUp(
            binaryMessenger: binaryMessenger,
            api: self
        )
    }
}

File: android/src/main/kotlin/com/inappstory/inappstory_plugin/adaptors/IASStatisticsManagerAdaptor.kt
package com.inappstory.inappstory_plugin.adaptors

import InAppStoryStatManagerHostApi
import com.inappstory.sdk.externalapi.InAppStoryAPI
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding

class IASStatisticsManagerAdaptor(
    flutterPluginBinding: FlutterPluginBinding,
    private val inAppStoryApi: InAppStoryAPI
) : InAppStoryStatManagerHostApi {
    init {
        InAppStoryStatManagerHostApi.setUp(flutterPluginBinding.binaryMessenger, this)
    }

    override fun sendStatistics(
        enabled: Boolean,
        callback: (Result<Unit>) -> Unit
    ) {
        inAppStoryApi.settings.sendStatistic(enabled)
        callback.invoke(Result.success(Unit))
    }
}